--- /home/users/hunter/backup/stored.c~	Fri Nov  2 07:29:03 2001
+++ src/stored.c	Sat Jun  1 09:15:37 2002
@@ -193,8 +193,8 @@
                     closesocket(ns); 
 		    exit(0);
 		  }
-		  sprintf(Sfilename, "%s/store/%04x.s", store_dir, FileNo);
-		  sprintf(Ifilename, "%s/store/%04x.i", store_dir, FileNo);
+		  sprintf(Sfilename, "%s/store/%01x/%01x/%04x.s", store_dir, FileNo>>12, (FileNo>>8)%16,  FileNo);
+		  sprintf(Ifilename, "%s/store/%01x/%01x/%04x.i", store_dir, FileNo>>12,  (FileNo>>8)%16, FileNo);
 
 		  if ((Ifd = fopen(Ifilename, "rb+")) == NULL) {
 		    if ((Ifd = fopen(Ifilename, "wb+")) == NULL) {
