diff -urNp -x '*.orig' mnogosearch-3.3.14.org/src/Makefile.am mnogosearch-3.3.14/src/Makefile.am
--- mnogosearch-3.3.14.org/src/Makefile.am	2013-03-30 15:10:18.000000000 +0100
+++ mnogosearch-3.3.14/src/Makefile.am	2021-10-05 22:44:49.937016015 +0200
@@ -45,9 +45,25 @@ dbug.c string.c date.c suggest.c groupby
 idn.c features.c message.c excerpt.c parsedocx.c parsertf.c
 
 EXTRA_libmnogosearch_la_SOURCES = efence.c ef_page.c ef_print.c
-libmnogosearch_la_LIBADD = @efOBJ@ -L. -L.libs -lmnogocharset
+libmnogosearch_la_LIBADD = @efOBJ@ libmnogocharset.la \
+	@VIRT_LFLAGS@ \
+	@SOLID_LFLAGS@ \
+	@UNIXODBC_LFLAGS@ \
+	@DB2_LFLAGS@ \
+	@IODBC_LFLAGS@ \
+	@EASYSOFT_LFLAGS@ \
+	@MYSQL_LFLAGS@ \
+	@PGSQL_LFLAGS@ \
+	@IBASE_LFLAGS@ \
+	@CTLIB_LFLAGS@ \
+	@SQLITE_LFLAGS@ \
+	@SQLITE3_LFLAGS@ \
+	@ORACLE8_LFLAGS@ \
+	@SAPDB_LFLAGS@ \
+	@SSL_LFLAGS@ \
+	@CHASEN_LFLAGS@
 libmnogosearch_la_LDFLAGS = -release 3.3
-libmnogosearch_la_DEPENDENCIES = libmnogocharset.la @efOBJ@
+libmnogosearch_la_DEPENDENCIES = @efOBJ@
 
 libmnogocharset_la_SOURCES = uconv.c unidata.c sgml.c unicode.c \
 uconv-8bit.c uconv-alias.c uconv-big5.c uconv-eucjp.c uconv-euckr.c \
@@ -82,20 +98,4 @@ INCLUDES = -I$(srcdir)/../include \
 @CHASEN_INCLUDE@
 
 LIBS = @STATIC_LFLAGS@ \
-@LIBS@ \
-@VIRT_LFLAGS@ \
-@SOLID_LFLAGS@ \
-@UNIXODBC_LFLAGS@ \
-@DB2_LFLAGS@ \
-@IODBC_LFLAGS@ \
-@EASYSOFT_LFLAGS@ \
-@MYSQL_LFLAGS@ \
-@PGSQL_LFLAGS@ \
-@IBASE_LFLAGS@ \
-@CTLIB_LFLAGS@ \
-@SQLITE_LFLAGS@ \
-@SQLITE3_LFLAGS@ \
-@ORACLE8_LFLAGS@ \
-@SAPDB_LFLAGS@ \
-@SSL_LFLAGS@ \
-@CHASEN_LFLAGS@
+@LIBS@
