--- mnogosearch-3.3.7/src/Makefile.am.orig	2007-09-01 17:52:27.000000000 +0200
+++ mnogosearch-3.3.7/src/Makefile.am	2008-09-18 20:03:05.211613660 +0200
@@ -38,9 +38,25 @@
 mirror.c sqldbms.c chinese.c parsexml.c wild.c hash.c segment.c \
 sql-mysql.c sql-oracle.c sql-ibase.c sql-pgsql.c
 EXTRA_libmnogosearch_la_SOURCES = efence.c ef_page.c ef_print.c
-libmnogosearch_la_LIBADD = @efOBJ@ -L. -L.libs -lmnogocharset
+libmnogosearch_la_LIBADD = @efOBJ@ libmnogocharset.la \
+	@VIRT_LFLAGS@ \
+	@SOLID_LFLAGS@ \
+	@UNIXODBC_LFLAGS@ \
+	@DB2_LFLAGS@ \
+	@IODBC_LFLAGS@ \
+	@EASYSOFT_LFLAGS@ \
+	@MYSQL_LFLAGS@ \
+	@PGSQL_LFLAGS@ \
+	@IBASE_LFLAGS@ \
+	@CTLIB_LFLAGS@ \
+	@SQLITE_LFLAGS@ \
+	@SQLITE3_LFLAGS@ \
+	@ORACLE8_LFLAGS@ \
+	@SAPDB_LFLAGS@ \
+	@SSL_LFLAGS@ \
+	@CHASEN_LFLAGS@
 libmnogosearch_la_LDFLAGS = -release 3.3
-libmnogosearch_la_DEPENDENCIES = libmnogocharset.la @efOBJ@
+libmnogosearch_la_DEPENDENCIES = @efOBJ@
 
 libmnogocharset_la_SOURCES = uconv.c unidata.c sgml.c unicode.c \
 uconv-8bit.c uconv-alias.c uconv-big5.c uconv-eucjp.c uconv-euckr.c \
@@ -75,20 +91,4 @@
 @CHASEN_INCLUDE@
 
 LIBS = @STATIC_LFLAGS@ \
-@LIBS@ \
-@VIRT_LFLAGS@ \
-@SOLID_LFLAGS@ \
-@UNIXODBC_LFLAGS@ \
-@DB2_LFLAGS@ \
-@IODBC_LFLAGS@ \
-@EASYSOFT_LFLAGS@ \
-@MYSQL_LFLAGS@ \
-@PGSQL_LFLAGS@ \
-@IBASE_LFLAGS@ \
-@CTLIB_LFLAGS@ \
-@SQLITE_LFLAGS@ \
-@SQLITE3_LFLAGS@ \
-@ORACLE8_LFLAGS@ \
-@SAPDB_LFLAGS@ \
-@SSL_LFLAGS@ \
-@CHASEN_LFLAGS@
+@LIBS@
