--- mnogosearch-3.2.39/src/Makefile.am.orig	2006-05-25 07:26:12.000000000 +0200
+++ mnogosearch-3.2.39/src/Makefile.am	2006-08-24 15:45:56.845971000 +0200
@@ -38,9 +38,9 @@
 mirror.c sqldbms.c chinese.c parsexml.c wild.c hash.c \
 sql-mysql.c sql-oracle.c sql-ibase.c sql-pgsql.c
 EXTRA_libmnogosearch_la_SOURCES = efence.c ef_page.c ef_print.c
-libmnogosearch_la_LIBADD = @efOBJ@ -L. -L.libs -lmnogocharset
+libmnogosearch_la_LIBADD = @efOBJ@ -L. libmnogocharset.la
 libmnogosearch_la_LDFLAGS = -release 3.2
-libmnogosearch_la_DEPENDENCIES = libmnogocharset.la @efOBJ@
+libmnogosearch_la_DEPENDENCIES = @efOBJ@
 
 libmnogocharset_la_SOURCES = uconv.c unidata.c sgml.c unicode.c \
 uconv-8bit.c uconv-alias.c uconv-big5.c uconv-eucjp.c uconv-euckr.c \
