--- /home/users/hunter/backup/Makefile.am~	Wed May 15 13:36:46 2002
+++ Makefile.am	Sun Jun  2 17:25:17 2002
@@ -5,21 +5,21 @@
 # copy dirs w/out Makefile.am into dist
 dist-hook:
 #	mkdir $(distdir)/include
-	cp -r $(srcdir)/include/ $(distdir)/include/
+	cp -r $(srcdir)/include/ $(DESTDIR)$(distdir)/include/
 	rm -f $(distdir)/include/udm_config.h $(distdir)/include/stamp.h 
-	mkdir $(distdir)/create
-	cp -r $(srcdir)/create/ $(distdir)/
-	mkdir $(distdir)/misc
-	cp -r $(srcdir)/misc/ $(distdir)/
+	mkdir $(DESTDIR)$(distdir)/create
+	cp -r $(srcdir)/create/ $(DESTDIR)$(distdir)/
+	mkdir $(DESTDIR)$(distdir)/misc
+	cp -r $(srcdir)/misc/ $(DESTDIR)$(distdir)/
 
 install-data-local:
-	$(mkinstalldirs) $(localstatedir)
-	$(mkinstalldirs) $(localstatedir)/raw/
-	$(mkinstalldirs) $(localstatedir)/splitter/
-	$(mkinstalldirs) $(localstatedir)/tree/
-	$(mkinstalldirs) $(localstatedir)/cache/
-	$(mkinstalldirs) $(localstatedir)/store/
-	chmod 777 $(localstatedir)/cache/
+	$(mkinstalldirs) $(DESTDIR)$(localstatedir)
+	$(mkinstalldirs) $(DESTDIR)$(localstatedir)/raw/
+	$(mkinstalldirs) $(DESTDIR)$(localstatedir)/splitter/
+	$(mkinstalldirs) $(DESTDIR)$(localstatedir)/tree/
+	$(mkinstalldirs) $(DESTDIR)$(localstatedir)/cache/
+	$(mkinstalldirs) $(DESTDIR)$(localstatedir)/store/
+	chmod 777 $(DESTDIR)$(localstatedir)/cache/
 
 uninstall-local:
 	-rmdir $(localstatedir)/raw/
